api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1184,6 +1184,14 @@ message Cluster {
   // If ``connection_pool_per_downstream_connection`` is true, the cluster will use a separate
   // connection pool for every downstream connection
   bool connection_pool_per_downstream_connection = 51;
+
+  // The idle timeout for connections. The idle timeout is defined as the period in which
+  // there are no active stream. When the idle timeout is reached the connection will be closed.
+  //
+  // .. warning::
+  //   Disabling this timeout has a highly likelihood of yielding connection leaks due to lost TCP
+  //   FIN packets, etc.
+  google.protobuf.Duration tcp_pool_idle_timeout = 57;
```

Comment:
If this is TCP only I think it should be in tcp-specific config, kind of like the upstream http protocol options.
I think it makes sense as a general field, though we'd need to put some thought into how it interacts with prefetching connections.

Reply:
tcp pool is used by several L7 protocol like thrift so I'm with you to put as a general field. 

Default idle timeout (with runtime guard unblock) would be an hour, and the timer starts ticking only if no one holds the connection. Does prefetching in this context mean creating at most 3 more connections for every new connection? If yes, I wouldn't worry about time performance for  1 hour idle timeout. That hurts only if we don't need that much connection every moment in previous hour.

Reply:
sorry @mattklein123 can I get your thoughts on this comment?
We have an HTTP equivalent and it's in the http-specific upstream config.  If we're going to leave this here I think we either have to deprecate the HTTP specific option, or at least mention the fact this field is ignored for HTTP upstreams.

Reply:
Oh I see sorry I missed this. Yeah can we just have a TCP protocol specific options much like we do for HTTP?